util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java

```diff
@@ -373,8 +373,7 @@ private static String getKey(final String rawKey, final CacheType cacheType, fin
                     // Make sure to re-hydrate the objects first (especially needed for create calls)
                     final Collection<M> reHydratedEntities = new ArrayList<>(entityRecordIds.size());
                     if (deletedAndUpdatedEntities.isEmpty()) {
-                        Preconditions.checkState(entityRecordIds.size() == 1 && changeType == ChangeType.INSERT, "Unexpected number of entityRecordIds=%s and changeType=%s", entityRecordIds, changeType);
-                        reHydratedEntities.add(sqlDao.getByRecordId(entityRecordIds.get(0), context));
+                        reHydratedEntities.addAll((List<M>) sqlDao.getByRecordIds(entityRecordIds, context));
```

Comment:
Out of curiosity -- why this change (just curious why the `Preconditions` has changed)? Is that for bulk insert?

Reply:
As discussed, bulk invoice item creation.